llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -18011,9 +18011,10 @@ bool AArch64TargetLowering::shouldFoldConstantShiftPairToMask(
       if (auto ShouldADD = *N->user_begin();
           ShouldADD->getOpcode() == ISD::ADD && ShouldADD->hasOneUse()) {
         if (auto ShouldLOAD = dyn_cast<LoadSDNode>(*ShouldADD->user_begin())) {
-          unsigned ByteVT = ShouldLOAD->getMemoryVT().getSizeInBits() / 8;
+          unsigned ByteVT = ShouldLOAD->getMemoryVT().getScalarSizeInBits() / 8;
```

Comment:
Because there are no tests to the contrary I've assumed the original intent of this code was to cover scalar types only, which also happens to be what we want for scalable vector loads.